airbyte-integrations/connectors/source-s3/source_s3/source_files_abstract/stream.py

```diff
@@ -165,143 +158,35 @@ def get_time_ordered_file_infos(self) -> List[FileInfo]:
-                # this compares datatype of every column that the two schemas have in common
-                for col in column_superset:
-                    if (col in master_schema.keys()) and (col in this_schema.keys()) and (master_schema[col] != this_schema[col]):
-                        # If this column exists in a provided schema or schema state, we'll WARN here rather than throw an error.
-                        # This is to allow more leniency as we may be able to coerce this datatype mismatch on read according to
-                        # provided schema state. Else we're inferring the schema (or at least this column) from scratch, and therefore
-                        # we try to choose the broadest type among two if possible
-                        broadest_of_types = self._broadest_type(master_schema[col], this_schema[col])
-                        type_explicitly_defined = col in self._schema.keys()
-                        override_type = broadest_of_types and not type_explicitly_defined
-                        if override_type:
-                            master_schema[col] = broadest_of_types
-                        if override_type or type_explicitly_defined:
-                            LOGGER.warn(
-                                f"Detected mismatched datatype on column '{col}', in file '{file_info}'. "
-                                + f"Should be '{master_schema[col]}', but found '{this_schema[col]}'. "
-                                + f"Airbyte will attempt to coerce this to {master_schema[col]} on read."
-                            )
-                            continue
-                        # otherwise throw an error on mismatching datatypes
-                        raise S3Exception(
-                            processed_files,
-                            "Column type mismatch",
-                            f"Detected mismatched datatype on column '{col}', in file '{file_info}'. "
-                            + f"Should be '{master_schema[col]}', but found '{this_schema[col]}'.",
-                            failure_type=FailureType.config_error,
-                        )
-
-                # missing columns in this_schema doesn't affect our master_schema, so we don't check for it here
-
-                # add to master_schema any columns from this_schema that aren't already present
-                for col, datatype in this_schema.items():
-                    if col not in master_schema.keys():
-                        master_schema[col] = datatype
-
-            LOGGER.info(f"determined master schema: {master_schema}")
-            self.master_schema = master_schema
-
-        return self.master_schema
+    @cached_property
+    def _auto_inferred_schema(self) -> Dict[str, Any]:
+        file_reader = self.fileformatparser_class(self._format)
+        file_info_iterator = iter(list(self.get_time_ordered_file_infos()))
```

Comment:
Please mention in the documentation that the "first" file we consider is the oldest one written to the prefix.